vector-im/element-android
# Fixes ended poll voting

## Type of change


## Content



Fixes polls being votable after being ended

## Motivation and context



Closes https://github.com/vector-im/element-android/issues/5040

The cause of the issue is found in `EventRelationsAggregationProcessor:388`.

When voting, the SDK checks that the eventTimestamp is before the closed time and if it is, the vote response can go through, despite the poll having already been ended (determined by the non-null presence of a closed time).

The fix on the SDK was to simply remove that extra condition. I checked that this shouldn't have any side-effects. I tried to approach this differently, i.e. trying to fix the event timestamp (`event.originServerTs`) set during the poll ending so that it matches the poll closed time but to no avail.

Though I should be able to investigate this with a bit more time if that's the preferrable course of action

I also added an extra layer of protection on the client side to check that the poll is in an ended state.

## Screenshots / GIFs



## Tests



- Set your device system time to 1 hour ahead of current time
- In any timeline, create a new poll
- Immediately end it
- Try to vote on it and see that it cannot be voted

## Tested devices

- OS version(s): Android 10

## Checklist
